administrator/components/com_workflow/src/Controller/GraphController.php

```diff
@@ -0,0 +1,307 @@
+
+            $model->setState('filter.workflow_id', $workflowId);
+            $model->setState('list.limit', 0);
+
+            $transitions = $model->getItems();
+            $response = [];
+
+            foreach ($transitions as $transition) {
+                $response[] = [
+                    'id'            => (int) $transition->id,
+                    'title'         => $transition->title,
+                    'description'   => $transition->description,
+                    'published'     => (bool) $transition->published,
+                    'from_stage_id' => (int) $transition->from_stage_id,
+                    'to_stage_id'   => (int) $transition->to_stage_id,
+                    'ordering'      => (int) $transition->ordering,
+                    'workflow_id'   => (int) $transition->workflow_id
+                ];
+            }
+
+            echo new JsonResponse($response);
+        } catch (\Exception $e) {
+            echo new JsonResponse($e->getMessage(), 'error', true);
+        }
+
+        $this->app->close();
+    }
+
+    /**
+     * Method to save stage positions
+     *
+     * @return  boolean  True if successful, false otherwise.
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function save()
+    {
+        // Check for request forgeries
+        $this->checkToken();
+
+        $app = $this->app;
+        $input = $app->input;
+        $workflowId = $input->getInt('id');
+        $extension = $this->input->getCmd('extension');
+        $options = $input->getCmd('option');
+
+        $task = $this->getTask();
+
+        $positionsJson = $input->getString('positions', '');
+        $positions = [];
+
+        if (!empty($positionsJson)) {
+            try {
+                $positions = json_decode($positionsJson, true);
+            } catch (\Exception $e) {
+                $app->enqueueMessage($e->getMessage(), 'error');
+            }
+        }
+
+        // Update positions if we have data
+        if (!empty($positions)) {
+            $model = $this->getModel('Stages');
+            $model->updatePositions($positions);
+            $app->enqueueMessage(Text::_('COM_WORKFLOW_POSITIONS_SAVED'));
+        }
+
+        // Redirect based on task
+        switch ($task) {
+            case 'apply':
+                $this->setRedirect(
+                    Route::_('index.php?option=' . $options . '&view=graph&id=' . $workflowId . '&extension=' . $extension, false)
+                );
+                break;
+
+            default:
+                $this->setRedirect(
+                    Route::_('index.php?option=' . $options . '&view=workflows&extension=' . $extension, false)
+                );
+                break;
+        }
+
+        return true;
+    }
+
+    /**
+     * Function to apply changes to the workflow graph
+     *
+     * @return  boolean  True if successful
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function apply()
```

Comment:
Not needed, just register an alias in the contructor: https://github.com/joomla/joomla-cms/blob/6.0-dev/libraries/src/MVC/Controller/FormController.php#L153